ruby/lib/clone_handler.rb

```diff
@@ -0,0 +1,136 @@
+require 'json'
+require 'net/http'
+
+class CloneHandler
+  PER_PAGE = 30.freeze
+  BASE_URL = 'https://api.github.com'.freeze
+  NETWORK_ERROR_RESPONSE = { 'status' => '999', 'message' => 'Network error' }.freeze
+
+  def self.get_username
+    Kernel.puts 'Please enter your Github username'
+    STDOUT.flush
+    STDIN.gets.chomp
+  end
+
+  def initialize(username)
+    @username = username
+    @last_repo_info_request_failed = false
+    @errors = []
+  end
+
+  def clone_all
+    validate_clonable
+    if errors?
+      print_errors
+      return false
+    end
+
+    Kernel.system(string_to_clone_all_repos_as_bash_job)
+  end
+
+  private
+
+  attr_reader :username, :errors
+  attr_accessor :last_repo_info_request_failed
+
+  def string_to_clone_all_repos_as_bash_job
+    clone_commands.flatten.join(' & ')
+  end
+
+  def clone_commands
+    repo_info.map { |info| build_repo_clone_command(info) }
+  end
+
+  def repo_info
+    @repo_info ||= page_count.times.map do |index|
+      next if last_repo_info_request_failed
```

Comment:
`next` should be `break`.

Reply:
I tried that. The map ends up being nil. Example:

```
 x = [1,2,3,4]
 y = x.map { |z| z * 2 } # =>  [2, 4, 6, 8]
 y = x.map { |z| z == 2 ? break : z * 2 } # => nil
```

I might try replacing this with a reduce or manual for loop so I can properly break.